portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,60 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+    
   @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String userName = request.getParameter("user-name");
```

Comment:
Maybe add a check to make sure the username and comment is not null.

Reply:
I managed to do this, but it was a little less intuitive than I thought. Is there anyway to prevent an html input form from submitting, given a null input? The best I managed to do was catch the null input in the servlet, then refreshing the page instead of sending the data to the datastore.

Reply:
I think you can do it in the javascript, to check the input string, if it is null, you can show an alert window to notice user.